crates/cadenza-eval/src/ir/types.rs

```diff
@@ -264,7 +264,8 @@ impl std::fmt::Display for IrInstr {
             IrInstr::Const {
                 result, ty, value, ..
             } => {
-                write!(f, "{}: {} = const {}", result, ty, value)
+                // For constants with type annotation: let v0: integer = 42
+                write!(f, "let {}: {} = {}", result, ty, value)
```

Comment:
I think we still should wrap the constant value in a `const` call, so `const 42`